assets/tooltip.css

```diff
@@ -1,95 +1,66 @@
+.tooltipArrow::before {
+  @apply border-solid block m-auto;
+  content: "";
+  height: 0;
+  width: 0;
 }
 
-.tooltip_topEnd::after {
-  content: " ";
-  border-width: 10px;
-  border-color: white transparent transparent transparent;
-  right: 10px;
-  @apply border-solid absolute top-full;
-}
-
-.tooltip_bottomCenter::after {
-  content: " ";
-  margin-left: -10px;
-  border-width: 10px;
-  border-color: transparent transparent white transparent;
-  @apply border-solid absolute bottom-full left-half;
+.tooltipArrow::after {
+  @apply border-solid block m-auto absolute;
+  content: "";
+  height: 0;
+  width: 0;
 }
 
-.tooltip_bottomStart::after {
-  content: " ";
-  border-width: 10px;
-  border-color: transparent transparent white transparent;
-  left: 10px;
-  @apply border-solid absolute bottom-full;
+.tooltipArrow-top {
+  @apply left-0;
+  margin-top: -10px;
+  top: 10px;
 }
 
-.tooltip_bottomEnd::after {
-  content: " ";
-  border-width: 10px;
+.tooltipArrow-top::after {
   border-color: transparent transparent white transparent;
-  right: 10px;
-  @apply border-solid absolute bottom-full;
+  border-width: 0 10px 10px 10px;
 }
 
-.tooltip_leftCenter::after {
-  content: " ";
-  margin-top: -10px;
-  border-width: 10px;
-  border-color: transparent transparent transparent white;
-  @apply border-solid absolute left-full top-half;
+.tooltipArrow-left {
+  @apply left-0;
+  margin-left: -10px;
 }
 
-.tooltip_leftStart::after {
-  content: " ";
-  top: 10px;
-  border-width: 10px;
-  border-color: transparent transparent transparent white;
-  @apply border-solid absolute left-full;
+.tooltipArrow-left::after {
+  @apply top-0;
+  border-color: transparent white transparent transparent;
+  border-width: 10px 10px 10px 0;
+  right: -5px;
 }
 
-.tooltip_leftEnd::after {
-  content: " ";
+.tooltipArrow-bottom {
+  @apply left-0;
+  margin-bottom: -20px;
   bottom: 10px;
-  border-width: 10px;
-  border-color: transparent transparent transparent white;
-  @apply border-solid absolute left-full;
 }
 
-.tooltip_rightCenter::after {
-  content: " ";
-  margin-top: -10px;
-  border-width: 10px;
-  border-color: transparent white transparent transparent;
-  @apply border-solid absolute right-full top-half;
+.tooltipArrow-bottom::after {
+  border-color: white transparent transparent transparent;
+  border-width: 10px 10px 0 10px;
+  bottom: 10px;
 }
 
-.tooltip_rightStart::after {
-  content: " ";
-  border-width: 10px;
-  top: 10px;
-  border-color: transparent white transparent transparent;
-  @apply border-solid absolute right-full;
+.tooltipArrow-right {
+  @apply right-0;
+  margin-right: -10px;
 }
 
-.tooltip_rightEnd::after {
-  content: " ";
-  border-width: 10px;
-  bottom: 10px;
-  border-color: transparent white transparent transparent;
-  @apply border-solid absolute right-full;
+.tooltipArrow-right::after {
+  @apply top-0;
+  border-color: transparent transparent transparent white;
+  border-width: 10px 0 10px 10px;
+  left: -5px;
 }
```

Comment:
can you quickly state what's holding us back from doing these with tailwind? this adds to the global CSS bundle while we have some parallel initiatives to reduce that size. it turns out to be pretty hard to build it back to tailwind once we have custom CSS

Reply:
As far as I know, it's not possible to style `after` pseudo-elements with tailwind.

Reply:
It's a bit hard to see in the diff but would it be possible to only keep the before/after classes in the global CSS to minimize it and use tailwind for the non-pseudo classes?

Reply:
My intention was to not add extra values to tailwind config since they are only used for the tooltip. Happy to change that if we prefer tailwind.